### ISSUE TITLE
Failed to connect to localhost port 9515: Connection refused

### ISSUE THREAD

**Issue by tyler36** [duncan3dc/dusk#2]
Hi,

I installed the package in a ZEND project and created a TEST file as describe on the 'Setup' page but I get the following error: 
```

Failed to connect to localhost port 9515: Connection refused
...\vendor\facebook\webdriver\lib\Remote\HttpCommandExecutor.php:286
...\vendor\facebook\webdriver\lib\Remote\RemoteWebDriver.php:535
...\vendor\facebook\webdriver\lib\Remote\RemoteWebDriver.php:277
...\vendor\laravel\dusk\src\Browser.php:298
...\vendor\duncan3dc\dusk\src\Dusk.php:102

Caused by
Facebook\WebDriver\Exception\WebDriverCurlException: Curl error thrown for http POST to /session/5cc69bc2db0bec0df1ac8169be1c3a47/url with params: {"url":"http:\/\/laravel.com"}

Failed to connect to localhost port 9515: Connection refused
```

A Chrome window does pop up but closes out within a second. Any ideas on what I am doing wrong?

OS: Win10

**Comment by duncan3dc** [comment]
Hi @tyler36, if you're seeing the Chrome window open then that rules out most of the usual stuff (permissions, wrong binary, etc).

Maybe you could get some more information by running the driver manually:
`./vendor/laravel/dusk/bin/chromedriver-win.exe --verbose`

**Comment by tyler36** [comment]
Hi @duncan3dc 
Sorry for the late response ... end of month craziness.

If I run the following command in a separate terminal before my tests, they seem to run fine:
```
"./vendor/laravel/dusk/bin/chromedriver-win.exe" --verbose
```

I see the following
```
Starting ChromeDriver 2.28.455520 (cc17746adff54984afff480136733114c6b3704b) on port 9515
Only local connections are allowed.
````

I do get a line just before the browser launch command (am using a Japanese keyboard):
```
[221.915][INFO]: Cannot switch to US keyboard layout - some keys may be interpreted incorrectly
```

**Comment by SimZal** [comment]
Hello, I am also using Win10 and I get a similar message. Manually starting chromedriver-win.exe resolves the issue. 
Original Dusk runs without problems for me, so I don't quite understand why it fails to start chromedriver. 

```
[17-May-2017 22:38:26 America/Los_Angeles] PHP Fatal error:  Uncaught Facebook\WebDriver\Exception\WebDriverCurlException: Curl error thrown for http POST to /session/6f8c510402b8295bd74e99ead8e8d912/url with params: {"url":"http:\/\/example.com"}

Failed to connect to localhost port 9515: Connection refused in ....\vendor\facebook\webdriver\lib\Remote\HttpCommandExecutor.php:286
Stack trace:
#0 ....\vendor\facebook\webdriver\lib\Remote\RemoteWebDriver.php(535): Facebook\WebDriver\Remote\HttpCommandExecutor->execute(Object(Facebook\WebDriver\Remote\WebDriverCommand))
#1 ....\vendor\facebook\webdriver\lib\Remote\RemoteExecuteMethod.php(40): Facebook\WebDriver\Remote\RemoteWebDriver->execute('get', Array)
#2 ....\vendor\facebook\webdriver\lib\WebDriverNavigation.php(84): Facebook\WebDriver\Remote\RemoteExecuteMethod->execute('get', Array)
#3 ....\vendor\laravel\dusk\src\Browser.php(111): Facebook\WebDriver\WebDriverNavigation->to('http://example....')
#4 ....\src\Dusk.ph in ....\vendor\facebook\webdriver\lib\Remote\HttpCommandExecutor.php on line 286
```

BTW: thanks for making this package, it looks very useful.

**Comment by SimZal** [comment]
Here's what worked for me:

The problem seems to be with Dusk.php __construct, which after completing, destroys the variable $driver (because it's not referenced anymore). That probably calls its destructor method that closes the external programs (chrome).
The solution was easy: add `$this->driver = $driver` at the end of __construct.

**Comment by tyler36** [comment]
@SimZal Confirmed.

After adding line to ```vendor\duncan3dc\dusk\src\dusk.php```, it worked as expected.
```
    public function __construct(DriverInterface $driver = null)
    {
        if ($driver === null) {
            $driver =  new Chrome;
        }
       ...
        **$this->driver = $driver;**
    }
```

Working with ```duncan3dc/dusk``` 0.2 & 0.3

OS: Win10
Composer:
        "php": ">=5.3.3",
        "duncan3dc/dusk": "^0.2.0",
        "phpunit/phpunit": "^5.7"

**Comment by duncan3dc** [comment]
That's strange, if that's the case then it shouldn't ever work.
I'll take a look at putting a fix in for it, thanks for the info

**Comment by SimZal** [comment]
@duncan3dc I also encountered this problem on Ubuntu. It worked without the fix on some systems, but it failed on others. I don't know why.

**Comment by tyler36** [comment]
@duncan3dc Just check my Homestead session too.
Without line - fails
With line - pass

Vagrant 1.9.6
Homestead 5.4.0 
Homestead box 2.1.0 -> Ubuntu 16.04.2, PHP 7.1.3-3

Oh, host PC
OS: Win10
PHP 5.6.28


Could never figure out which was more frustrating: 
- something that should work but doesn't 
- shouldn't work but does ;)

**Comment by duncan3dc** [comment]
This is fixed in [0.3.1](https://github.com/duncan3dc/dusk/releases/tag/0.3.1) now, thanks for your help everybody

**Comment by duncan3dc** [comment]
Hi @jvillaruel, this is a separate project for using dusk without Laravel, we are not associated with Laravel in any way.

Please try https://laravel.com for help with using Laravel
